### PR TITLE
LEDs, Logging & Pins Update

### PR DESCRIPTION
**Pull request by lanador**
- Removed LED code from GPS file
- Logging module updates: added pre-flight check, added logging to .csv
- Updated pins for LEDs and thermistors

### CODE REVIEW

File: HAB/HAB.ino [hunk 1]
```diff
@@ -17,57 +17,61 @@ void setup() {
...
    Logging::logSystemData(GPS::getRawGPSData());
    Logging::logSystemData(String(GPS::getCurrentAltitude()));
  }
    delay(1000);
 }
+
+//Doors opening and closing at 60000 ft TODO: altitude is in meters, set threshold and pass it here
+
+const long altitudeOpenThreshold = 60000;
+const long altitudeCloseThreshold = 55000;
```
Comment (vitosamson) [inline HAB/HAB.ino]: For a future commit: let's move these to the header file as constant definitions.

File: HAB/HAB.ino [hunk 2]
```diff
@@ -17,57 +17,61 @@ void setup() {
...
+void adjustDoorIfNeeded(long altitude) {
+  Door::DoorStatus door1Status = Door::getDoorStatus1();
+  Door::DoorStatus door2Status = Door::getDoorStatus2();
+  if (altitude >= altitudeOpenThreshold && (door1Status == Door::DoorStatus::DOOR_CLOSED || door2Status == Door::DoorStatus::DOOR_CLOSED)){
```
Comment (vitosamson) [inline HAB/HAB.ino]: For a future commit: the `openDoor()` and `closeDoor()` functions check the door status, so we don't need to do that here.

File: HAB/HAB_GPS.ino [hunk 3]
```diff
@@ -42,10 +42,10 @@ bool init() {
...
 
       // Flash the GPS status LED to indicate that we're waiting for GPS acquisition
-      digitalWrite(GPS_STATUS_LED_PIN, HIGH);
-      delay(400);
-      digitalWrite(GPS_STATUS_LED_PIN, LOW);
-      delay(400);
+//      digitalWrite(GPS_STATUS_LED_PIN, HIGH);
+//      delay(400);
+//      digitalWrite(GPS_STATUS_LED_PIN, LOW);
+//      delay(400);
```
Comment (vitosamson) [inline HAB/HAB_GPS.ino]: Can we either remove these lines altogether, or add a comment above them explaining that they cause the GPS initialization to fail?

File: HAB/HAB_Logging.h [hunk 4]
```diff
@@ -1,9 +1,42 @@
...
+class MissionData {
+public:
+
+  static String getTitles();
+  String toString();
+  void setRawGpsData(const char* rawGpsData);
+  void setGpsAltitude (long gpsAltitude);
+  void setOnboardTemperature(Sensors::TemperatureData onboardTemperature);
+  void setOutdoorTopTemperature(Sensors::TemperatureData OutdoorTopTemperature);
+  void setOutdoorBottomTemperature(Sensors::TemperatureData OutdoorBottomTemperature);
```
Comment (vitosamson) [inline HAB/HAB_Logging.h]: Small nitpick: for lines 19 and 20, the variables (`outdoorTopTemperature` and `outdoorBottomTemperature`) should start with a lowercase letter.

File: HAB/HAB_Logging.ino [hunk 5]
```diff
@@ -32,16 +32,49 @@ bool init() {
...
+  // Files are created, set the title of the CSV and write on systemFile
+  missionFile.println(MissionData::getTitles());
+  systemFile.println("System File initializing...");
+
+  // Read files to check if data was recorded.
+  char firstMissionChar = missionFile.read();
+  char firstSystemChar = systemFile.read();
+  // Serial.println("Char is");
+  // Serial.println(firstSystemChar);
+  // Serial.println(firstMissionChar);
```
Comment (vitosamson) [inline HAB/HAB_Logging.ino]: Please remove these debug comments.

File: HAB/HAB_Logging.ino [hunk 6]
```diff
@@ -71,35 +104,35 @@ void logSystemData(String data) {
   // so we need to check if the SD card has been initialized yet before
   // writing to it.
   if (didInit) {
-    File file = SD.open(systemLogFileName);
+    File file = SD.open(systemLogFileName, FILE_WRITE);
     if (file) {
+      Serial.println("File exists");
```
Comment (vitosamson) [inline HAB/HAB_Logging.ino]: Remove debug logging.

File: HAB/HAB_Logging.ino [hunk 7]
```diff
@@ -71,35 +104,35 @@ void logSystemData(String data) {
...
-  // if (Serial) {
-  //   if (GPS::gpsReady()) {
-  //     String timestamp = createTimestamp();
-  //     Serial.println(timestamp + " " + data);
-  //   } else {
-  //     Serial.println(data);
-  //   }
-  // }
 
+ //TODO: rename logMissionData so they are not same
```
Comment (vitosamson) [inline HAB/HAB_Logging.ino]: I don't think this comment is relevant anymore?

File: HAB/HAB_Logging.ino [hunk 8]
```diff
@@ -71,35 +104,35 @@ void logSystemData(String data) {
...
+ 
+void logMissionData(String data) {
+  logSystemData("Logging missionData.");
```
Comment (vitosamson) [inline HAB/HAB_Logging.ino:123]: Remove debug logging.

File: HAB/HAB_Logging.ino [hunk 9]
```diff
@@ -71,35 +104,35 @@ void logSystemData(String data) {
...
   File file = SD.open(missionLogFileName, FILE_WRITE);
   if (file) {
     file.println(data);
     file.close();
+  } else {
+    Serial.println("Could not open file " + missionLogFileName);
```
Comment (vitosamson) [inline HAB/HAB_Logging.ino:129]: Let's change this to `logSystemData()` instead of `Serial.println()`.

**Review by vitosamson — CHANGES_REQUESTED** [review]
@lanador I left a couple comments which can be taken care of in a future commit, and a couple that should be cleaned up before merging this PR.

File: HAB/HAB_GPS.ino (hunk 3 above)
Comment (lanador) [reply HAB/HAB_GPS.ino]: Sorry about that. I think this is a mistaken commit. This code should have been removed and this commit is from the time when I commented it out.

**Review by vitosamson — APPROVED** [review]
🤘

**Pull request merged**